mac/KeymanEngine4Mac/KeymanEngine4Mac/CoreWrapper/CoreWrapper.m

```diff
@@ -169,8 +166,73 @@ -(BOOL)processVirtualKey:(uint16_t)keyCode
   return (result==KM_CORE_STATUS_OK);
 }
 
--(NSArray*)loadActionsForLastKeyProcessed {
-  return [self loadActionsUsingCore];
+-(CoreKeyOutput*)loadOutputForLastKeyProcessed {
+  return [self loadActionStructUsingCore];
+}
+
+-(CoreKeyOutput*)loadActionStructUsingCore {
+  [self.coreHelper logDebugMessage:@"CoreWrapper loadActionStructUsingCore"];
+  km_core_actions * actions = km_core_state_get_actions(self.coreState);
+  CoreKeyOutput *output = [self createCoreKeyOutputForActionsStruct:actions];
+
+  km_core_status result = km_core_actions_dispose(actions);
+  [self.coreHelper logDebugMessage:@"km_core_actions_dispose() result = %u\n", result];
+  return output;
+}
+
+-(CoreKeyOutput*) createCoreKeyOutputForActionsStruct:(km_core_actions*)actions {
+  NSString* text = [self.coreHelper utf32CStringToString:actions->output];
+  NSDictionary* options = [self convertOptionsArray:actions->persist_options];
+  CapsLockState capsLock = [self convertCapsLockState:actions->new_caps_lock_state];
+
+  CoreKeyOutput* coreKeyOutput = [[CoreKeyOutput alloc] init: actions->code_points_to_delete textToInsert:text optionsToPersist:options alert:actions->do_alert emitKeystroke:actions->emit_keystroke capsLockState:capsLock];
+
+  return coreKeyOutput;
+}
+
+-(NSDictionary*)convertOptionsArray:(km_core_option_item*)options {
+  NSDictionary* optionsDictionary = nil;
+  unichar const * optionsKey = options->key;
+  unichar const * optionsValue = options->value;
+  if ((optionsKey != nil) && (optionsValue != nil)) {
+    optionsDictionary = [[NSDictionary alloc] init];
+    NSString *key = [self.coreHelper createNSStringFromUnicharString:optionsKey];
+    NSString *value = [self.coreHelper createNSStringFromUnicharString:optionsValue];
+    [optionsDictionary insertValue:value inPropertyWithKey:key];
+  }
+/*
+  if (options) {
+    optionsDictionary = [[NSDictionary alloc] init];
+    for (; options->key != 0; ++options) {
+      unichar const * optionsKey = options->key;
+      unichar const * valueKey = options->value;
+
+      NSString *key = [self.coreHelper createNSStringFromUnicharString:optionsKey];
+      NSString *value = [self.coreHelper createNSStringFromUnicharString:valueKey];
+      [optionsDictionary insertValue:value inPropertyWithKey:key];
+    }
+  }
+ */
```

Comment:
```suggestion
```

Reply:
I'll clean this up after merging and testing with the core persist options fix